### DISCUSSION TITLE
Bind values to placeholders in word document

### DISCUSSION THREAD

**Discussion by inane** [dolanmiu/docx#1004] [General]
Hello,

I have known this module today and I am trying to find the possibility to bind values to placeholders specified in my word document. Is that possible? Currently I have named in the word document like $$name$$ and I would like to bind in that namespace my value.

Thank you in advance

**Comment by dolanmiu** [comment]
You can yes, by doing it on the JavaScript/TypeScript side with JavaScript string templates:

```
new Paragraph({
    text: `${myVar} is very good: ${myVar2}`,
});
```

Alternatively, try another library called `docx-templater`

**Reply by inane** [reply]
I have used that library but it has not free support for images. It is an additional paid module.

I will check your approach it is very interesting and versatile your library

**Reply by dolanmiu** [reply]
Thanks!

**Reply by inane** [reply]
And @dolanmiu  is it possible to use "readFileSync" function to get a docx with placeholders and bind those values?

Currently I am using Pizzip dependency from docx-templater but I don´t know with your library is possible without it.

Thank you!!

**Reply by dolanmiu** [reply]
nope that is not possible sorry

This library is not capable of reading and modifying word documents

**Reply by inane** [reply]
It is only possible to create a new document, right?  Is this feature possible in the future? 

With your knowledge, do you think is possible a workaround to read word documents using your library?

Thank you for your replies.

**Reply by dolanmiu** [reply]
Correct yes

only allows creation of documents
